query/src/sql/planner/plans/scalar.rs

```diff
@@ -31,6 +32,8 @@ pub trait ScalarExpr {
 
     fn used_columns(&self) -> ColumnSet;
 
+    fn is_deterministic(&self) -> bool;
```

Comment:
What is this for?

Reply:
cluster key expression must be deterministic, so I've to check the property.

Reply:
@andylokandy Nice. By the way, you may need to add some check about deterministic cluster key for `CREATE TABLE` as well, which I haven't done before.